Source/automap.cpp

```diff
@@ -40,21 +35,48 @@ enum MapColors : uint8_t {
 	MapColorsItem = (PAL8_BLUE + 1),
 };
 
-constexpr uint16_t MapFlagsType = 0x000F;
-/** these are in the second byte */
-enum MapFlags : uint8_t {
+/** The general shape of the tile */
+enum class AutomapTypes : uint8_t {
+	None,
+	Diamond,
+	Vertical,
+	Horizontal,
+	Cross,
+	FenceVertical,
+	FenceHorizontal,
+	Corner,
+	CaveHorizontalCross,
+	CaveVerticalCross,
+	CaveHorizontal,
+	CaveVertical,
+	CaveCross,
+};
+
+/** Additional details about the given tile */
+enum AutomapFlags : uint8_t {
 	// clang-format off
-	MapFlagsMapVerticalDoor   = 1 << 0,
-	MapFlagsMapHorizontalDoor = 1 << 1,
-	MapFlagsVerticalArch      = 1 << 2,
-	MapFlagsHorizontalArch    = 1 << 3,
-	MapFlagsVerticalGrate     = 1 << 4,
-	MapFlagsHorizontalGrate   = 1 << 5,
-	MapFlagsDirt              = 1 << 6,
-	MapFlagsStairs            = 1 << 7,
+	AutomapFlagVerticalDoor    = 1 << 0,
+	AutomapFlagHorizontalDoor  = 1 << 1,
+	AutomapFlagVerticalArch    = 1 << 2,
+	AutomapFlagHorizontalArch  = 1 << 3,
+	AutomapFlagVerticalGrate   = 1 << 4,
+	AutomapFlagHorizontalGrate = 1 << 5,
+	AutomapFlagDirt            = 1 << 6,
+	AutomapFlagStairs          = 1 << 7,
 	// clang-format on
 };

```

Comment:
Didn't know that was possible, will have to look it to it

Reply:
I wasn't 100% sure how it would behave while I was writing this review so I half-assed an implementation to test. Finished it off after dinner and opened #2534 against this branch if you wanted to use some of it :)